src/Features/Core/Portable/EditAndContinue/SemanticEditInfo.cs

```diff
@@ -10,20 +10,30 @@ namespace Microsoft.CodeAnalysis.EditAndContinue
     internal readonly struct SemanticEditInfo
     {
         /// <summary>
-        /// <see cref="SemanticEditKind.Insert"/> or <see cref="SemanticEditKind.Update"/>.
+        /// <see cref="SemanticEditKind.Insert"/> or <see cref="SemanticEditKind.Update"/> or <see cref="SemanticEditKind.Delete"/>.
         /// </summary>
         public SemanticEditKind Kind { get; }
 
         /// <summary>
         /// If <see cref="Kind"/> is <see cref="SemanticEditKind.Insert"/> represents the inserted symbol in the new compilation.
         /// If <see cref="Kind"/> is <see cref="SemanticEditKind.Update"/> represents the updated symbol in both compilations.
+        /// If <see cref="Kind"/> is <see cref="SemanticEditKind.Delete"/> represents the deleted symbol in the old compilation.
         /// 
         /// We use <see cref="SymbolKey"/> to represent the symbol rather then <see cref="ISymbol"/>,
         /// since different semantic edits might have been calculated against different solution snapshot and thus symbols are not directly comparable.
         /// When the edits are processed we map the <see cref="SymbolKey"/> to the current compilation.
         /// </summary>
         public SymbolKey Symbol { get; }
 
+        /// <summary>
+        /// If <see cref="Kind"/> is <see cref="SemanticEditKind.Delete"/> represents the containing symbol in the new compilation.
+        /// 
+        /// We use <see cref="SymbolKey"/> to represent the symbol rather then <see cref="ISymbol"/>,
+        /// since different semantic edits might have been calculated against different solution snapshot and thus symbols are not directly comparable.
+        /// When the edits are processed we map the <see cref="SymbolKey"/> to the current compilation.
+        /// </summary>
+        public SymbolKey NewSymbolKey { get; }
```

Comment:
>NewSymbolKey [](http://example.com/codeflow?start=25&length=12)

NewContainingSymbol?

Reply:
`Symbol` property is not tied to the old or new compilation. It's a `SymbolKey` that can be resolved in either old or new compilation depending on the edit. For example, for insert it refers to a symbol in the new compilation. For update it refers to both the new and the old symbol. 

Since `NewSymbolKey` is also `SymbolKey` it doesn't seem like it should refer to a specific compilation. For delete we need to represent 2 different symbols - one is in the old compilation (`Symbol`) and the other in the new compilation. How about `DeleteSymbolContainer`?

Reply:
Okay, I guess that makes sense too. We can represent two different symbols, but either one might be null depending on the edit itself.